dns_trace.go

```diff
@@ -0,0 +1,92 @@
+package thousandeyes
+
+import (
+	"fmt"
+)
+
+// DNSTrace - DNS trace test
+type DNSTrace struct {
+	Agents               []Agent        `json:"agents,omitempty"`
+	AlertsEnabled        int            `json:"alertsEnabled,omitempty"`
+	AlertRules           []AlertRule    `json:"alertRules,omitempty"`
+	APILinks             []ApiLink      `json:"apiLinks,omitempty"`
+	CreatedBy            string         `json:"createdBy,omitempty"`
+	CreatedDate          string         `json:"createdDate,omitempty"`
+	Description          string         `json:"description,omitempty"`
+	Enabled              int            `json:"enabled,omitempty"`
+	Groups               []GroupLabel   `json:"groups,omitempty"`
+	LiveShare            int            `json:"liveShare,omitempty"`
+	ModifiedBy           string         `json:"modifiedBy,omitempty"`
+	ModifiedDate         string         `json:"modifiedDate,omitempty"`
+	SavedEvent           int            `json:"savedEvent,omitempty"`
+	SharedWithAccounts   []AccountGroup `json:"sharedWithAccounts,omitempty"`
+	TestID               int            `json:"testId,omitempty"`
+	TestName             string         `json:"testName,omitempty"`
+	Type                 string         `json:"type,omitempty"`
+	DNSTransportProtocol string         `json:"dnsTransportProtocol,omitempty"`
+	Domain               string         `json:"domain,omitempty"`
+	Interval             int            `json:"interval,omitempty"`
+}
+
+// AddDNSTrace - Add dns trace test
+func (t *DNSTrace) AddDNSTrace(id int) {
```

Comment:
AddAgent ?

Reply:
yea, I am adding just test types here.  FWIW I think the test vs agent model of the API is going to lead to confusion w/ this SDK... Especially w/o including `Test` in the name of the tests... Not sure how you feel about that...

Reply:
![image](https://user-images.githubusercontent.com/58234/78153481-1b505a80-7409-11ea-97c3-156db2fe32f6.png)

Reply:
@william20111 you can see there is a type of test called "dns-trace"